tools/misc/tpm2_checkquote.c

```diff
@@ -160,57 +161,109 @@ static TPM2B_ATTEST *message_from_file(const char *msg_file_path) {
     return msg;
 }
 
-static bool pcrs_from_file(const char *pcr_file_path,
-        TPML_PCR_SELECTION *pcr_select, tpm2_pcrs *pcrs) {
-
-    bool result = false;
-    unsigned long size;
+static bool parse_selection_data_from_selection_string(FILE *pcr_input,
+    TPML_PCR_SELECTION *pcr_select, tpm2_pcrs *pcrs) {
 
-    if (!files_get_file_size_path(pcr_file_path, &size)) {
+    bool result = pcr_parse_selections(ctx.pcr_selection_string, pcr_select);
+    if (!result) {
+        LOG_ERR("Could not parse PCR selections");
         return false;
     }
 
-    if (!size) {
-        LOG_ERR("The pcr file \"%s\" is empty", pcr_file_path);
-        return false;
-    }
+    //TODO: The only possible value for tpm2_pcrs->count is 1
+    pcrs->count = 1;
 
-    FILE *pcr_input = fopen(pcr_file_path, "rb");
-    if (!pcr_input) {
-        LOG_ERR("Could not open PCRs input file \"%s\" error: \"%s\"",
-                pcr_file_path, strerror(errno));
-        goto out;
+    unsigned i = 0;
+    unsigned j = 0;
+    unsigned read_size = 0;
+    for (i = 0; i < pcr_select->count; i++) {
+        read_size = tpm2_alg_util_get_hash_size(pcr_select->pcrSelections[i].hash);
+
+        for (j = 0; j < pcr_select->pcrSelections[i].sizeofSelect * 8; j++) {
+            if ((pcr_select->pcrSelections[i].pcrSelect[j / 8] & 1 << (j % 8)) != 0) {
+                pcrs->pcr_values[0].digests[pcrs->pcr_values[0].count].size = read_size;
```

Comment:
I think scan-build is complaining because count can go beyond 8, which is the max size of field digests.

if pcr_select->count is 1, and sizeofSelect is 2,  count++ takes count to > 7. Additionally, if pcr_select->count is 2, on the second iteration of the loop count is also not reset causing the same issue.